File: UI/grooveon_desktop/lib/models/request/question_upsert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'question_upsert_request.g.dart';

@JsonSerializable()
class QuestionUpsertRequest {
  final int userId;
  final String title;
  final String content;
  final String status;
  final String? answer;

  QuestionUpsertRequest({
    required this.userId,
    required this.title,
    required this.content,
    this.status = "Pending",
    this.answer,
  });

  factory QuestionUpsertRequest.fromJson(Map<String, dynamic> json) =>
      _$QuestionUpsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$QuestionUpsertRequestToJson(this);
}

File: GrooveOn.Models/RequestObjects/QuestionUpsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace GrooveOn.Model.RequestObjects
{
    public class QuestionUpsertRequest
    {
        [Required]
        public int UserId { get; set; }

        [Required]
        [MinLength(3)]
        public string Title { get; set; } = string.Empty;

        [Required]
        [MinLength(5)]
        public string Content { get; set; } = string.Empty;

        public string Status { get; set; } = "Pending";

        public string? Answer { get; set; }
    }
}